archive/rpm.ksy

```diff
@@ -0,0 +1,1076 @@
+      -orig-id: RPMTAG_ENHANCENAME
+    5056:
+      id: enhance_version
+      -orig-id: RPMTAG_ENHANCEVERSION
+    5057:
+      id: enhance_flags
+      -orig-id: RPMTAG_ENHANCEFLAGS
+    5058:
+      id: recommend_nevrs
+      -orig-id: RPMTAG_RECOMMENDNEVRS
+    5059:
+      id: suggest_nevrs
+      -orig-id: RPMTAG_SUGGESTNEVRS
+    5060:
+      id: supplement_nevrs
+      -orig-id: RPMTAG_SUPPLEMENTNEVRS
+    5061:
+      id: enhance_nevrs
+      -orig-id: RPMTAG_ENHANCENEVRS
+    5062:
+      id: encoding
+      -orig-id: RPMTAG_ENCODING
+    5066:
+      id: file_trigger_scripts
+      -orig-id: RPMTAG_FILETRIGGERSCRIPTS
+    5067:
+      id: file_trigger_script_prog
+      -orig-id: RPMTAG_FILETRIGGERSCRIPTPROG
+    5068:
+      id: file_trigger_script_flags
+      -orig-id: RPMTAG_FILETRIGGERSCRIPTFLAGS
+    5069:
+      id: file_trigger_name
+      -orig-id: RPMTAG_FILETRIGGERNAME
+    5070:
+      id: file_trigger_index
+      -orig-id: RPMTAG_FILETRIGGERINDEX
+    5071:
+      id: file_trigger_version
+      -orig-id: RPMTAG_FILETRIGGERVERSION
+    5072:
+      id: file_trigger_flags
+      -orig-id: RPMTAG_FILETRIGGERFLAGS
+    5076:
+      id: trans_file_trigger_scripts
+      -orig-id: RPMTAG_TRANSFILETRIGGERSCRIPTS
+    5077:
+      id: trans_file_trigger_script_prog
+      -orig-id: RPMTAG_TRANSFILETRIGGERSCRIPTPROG
+    5078:
+      id: trans_file_trigger_script_flags
+      -orig-id: RPMTAG_TRANSFILETRIGGERSCRIPTFLAGS
+    5079:
+      id: trans_file_trigger_name
+      -orig-id: RPMTAG_TRANSFILETRIGGERNAME
+    5080:
+      id: trans_file_trigger_index
+      -orig-id: RPMTAG_TRANSFILETRIGGERINDEX
+    5081:
+      id: trans_file_trigger_version
+      -orig-id: RPMTAG_TRANSFILETRIGGERVERSION
+    5082:
+      id: trans_file_trigger_flags
+      -orig-id: RPMTAG_TRANSFILETRIGGERFLAGS
+    5084:
+      id: file_trigger_priorities
+      -orig-id: RPMTAG_FILETRIGGERPRIORITIES
+    5085:
+      id: trans_file_trigger_priorities
+      -orig-id: RPMTAG_TRANSFILETRIGGERPRIORITIES
+    5086:
+      id: file_trigger_conds
+      -orig-id: RPMTAG_FILETRIGGERCONDS
+    5087:
+      id: file_trigger_type
+      -orig-id: RPMTAG_FILETRIGGERTYPE
+    5088:
+      id: trans_file_trigger_conds
+      -orig-id: RPMTAG_TRANSFILETRIGGERCONDS
+    5089:
+      id: trans_file_trigger_type
+      -orig-id: RPMTAG_TRANSFILETRIGGERTYPE
+    5090:
+      id: file_signatures
+      -orig-id: RPMTAG_FILESIGNATURES
+    5091:
+      id: file_signature_length
+      -orig-id: RPMTAG_FILESIGNATURELENGTH
+    5092:
+      id: payload_digest
+      -orig-id: RPMTAG_PAYLOADDIGEST
+    5093:
+      id: payload_digest_algo
+      -orig-id: RPMTAG_PAYLOADDIGESTALGO
+    5096:
+      id: modularity_label
+      -orig-id: RPMTAG_MODULARITYLABEL
+    5097:
+      id: payload_digest_alt
+      -orig-id: RPMTAG_PAYLOADDIGESTALT
+  header_types:
+    # from LSB
+    0: not_implemented
+    1: char
+    2: int8
+    3: int16
+    4: int32
+    5: int64 # reserved
```

Comment:
I think that _Reserved_ in https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/pkgformat.html#AEN63192 rather means that it wasn't used by any RPM Tag yet at the time when the LSB RPM spec was up to date. This is no longer the case, there _are_ a few `int64` tags now (search for "`int64`" in https://github.com/rpm-software-management/rpm/blob/911448f2/doc/manual/tags.md) - Longarchivesize (271), Longsize (5009), Longfilesizes (5008) and Longsigsize (270).

Also `int64` is fully implemented in `rpm` (see https://github.com/rpm-software-management/rpm/search?q=RPM_INT64_TYPE), so I don't see reason not to implement it in the `.ksy`.

Reply:
LSB seems to be horribly outdated when it comes to RPM (also only talks about gzip compression, although there has been support for other formats for years) so it should be taken with a grain of salt. I agree with you that it can be changed. I missed it because the files I tested with didn't contain RPMs that are *that* big.

Reply:
Yes, rpm >= 4.6 uses int64 for large sizes and you will need to implement it to deal with real-world packages in todays world. LSB seems stuck in the last millenium when it comes to rpm. See https://rpm.org/devel_doc/large_files.html for documentation.